src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/util/LicenceKind.kt

```diff
@@ -4,25 +4,29 @@ import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus.
 import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus.VARIATION_IN_PROGRESS
 
 enum class LicenceKind(
-  val initialStatus: LicenceStatus,
-  val copyEventType: LicenceEventType,
-  val submittedEventType: LicenceEventType,
+  val initialStatus: () -> LicenceStatus,
+  val creationEventType: () -> LicenceEventType,
+  val copyEventType: () -> LicenceEventType,
+  val submittedEventType: () -> LicenceEventType,
```

Comment:
These are now lazy to allow state checking in the unlikely case of an unsupported transition